### PR TITLE
Pr880

### PR DESCRIPTION
**Pull request by egeerardyn**
This fixes #880.

### CODE REVIEW

**Comment by egeerardyn** [comment]
@matlab2tikzBot: test this PR

File: src/matlab2tikz.m [hunk 1]
```diff
@@ -1911,6 +1919,11 @@ function assertRegularAxes(handle)
             || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
         lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
     end
+
+    % print no lines
+    if isNone(lineStyle) || lineWidth==0
```
Comment (miscco) [inline src/matlab2tikz.m]: Dont we have this isVisibleLine function`?

File: src/matlab2tikz.m [hunk 2]
```diff
@@ -4082,16 +4105,27 @@ function warnFacetedInterp(m2t)
...
+        [m2t, table, tableOptions] = makeTable(m2t, '', xData, '', yData, '', zData);
+        % Print out
+        tabOpts  = opts_print(tableOptions);
+        str = sprintf('\\addplot3 [%s]\n table[%s] {%s};\n ', ...
+                         drawOpts, tabOpts, table);
+    else
+        [m2t, table, tableOptions] = makeTable(m2t, '', xData, '', yData);
+        % Print out
+        tabOpts  = opts_print(tableOptions);
+        str = sprintf('\\addplot[%s] plot table[%s] {%s};\n', ...
```
Comment (miscco) [inline src/matlab2tikz.m]: While you are at it didnt we want to use table here?

Comment (egeerardyn) [inline src/matlab2tikz.m]: I will tackle those in a separate PR: then it's just a question of search-and-replace-and-update-hashes.

File: src/matlab2tikz.m [hunk 3]
```diff
@@ -1473,14 +1474,21 @@ case guitypes()
             % axis styles (e.g., colors).
             options = opts_add(options, 'separate axis lines');
         end
```
Comment (miscco) [inline src/matlab2tikz.m]: I am unsure, if the chunck below is needed or correct. 

a) Is there a possibility to change the color in matlab
b) if there is such a possibility, why always use the same color
c) As far as i know setting YColor and friends worked fine.

File: src/matlab2tikz.m (hunk 2 above)
Comment (miscco) [inline src/matlab2tikz.m]: Fine with me :+1:

File: src/matlab2tikz.m (hunk 1 above)
Comment (egeerardyn) [inline src/matlab2tikz.m]: Indeed we do have such a function.
However, that is for `line` objects (with markers, deviations, ...), while this relates to the lines of an `axes` object. Probably we could use that function (it should be robust enough to handle it), I'm just not sure what the best is:
 - slightly abuse that function for more readable code, or
 - do explicit checks for slightly more complex code

Comment (miscco) [inline src/matlab2tikz.m]: Nah keep it as it is. Otherwise, there is always the possibiilty that a check fails etc.

Comment (egeerardyn) [inline src/matlab2tikz.m]: 👍

File: src/matlab2tikz.m (hunk 3 above)
Comment (egeerardyn) [inline src/matlab2tikz.m]: In #880 there are some figures to illustrate the behavior (which is where most of this code stems from).
I'm also not 100% convinced about this, but it matches MATLAB behavior better, which is what is expected from `strict` mode.

However, now I think about it: perhaps it is better to only add this fragment when the strict parameter is passed.

**Comment by egeerardyn** [comment]
@matlab2tikzBot: test this PR

**Comment by egeerardyn** [comment]
@matlab2tikzBot: test this PR

File: src/matlab2tikz.m [hunk 4]
```diff
@@ -4491,14 +4525,19 @@ function warnFacetedInterp(m2t)
     pgfplotsColormap = sprintf('colormap={%s}{[1%s] %s}',name, unit, join(m2t, colSpecs, '; '));
 end
 % ==============================================================================
-function fontStyle = getFontStyle(m2t, handle)
+function [m2t, fontStyle] = getFontStyle(m2t, handle)
```
Comment (miscco) [inline src/matlab2tikz.m]: What i do not like here is the way we handle fontStyle. Would it be possible to use the opts array we have here too? That would make the code less special.

Comment (egeerardyn) [inline src/matlab2tikz.m]: Sorry for the late response. I think I see what you mean: this function basically translates the MATLAB font spec to LaTeX by building a string to represent it. However, that is unrelated to option arrays.
So, no, we cannot use opts arrays here.

**Comment by miscco** [comment]
Damn, i just spend some hours on Thursday to find out how to display the label of an invisible axis. 
:+1: from me i think it is definitely better to add that now.

**Comment by egeerardyn** [comment]
Okay, I'm merging this as-is. It is not perfect and I have noticed that e.g. on R2014b, `ACID(42)` does not compile (legends in bode plots) well, but this is probably not the fault of this PR anyway. Everything else in the graphical test (only R2014b executed, though), looks fine; and everything passes in the headless test on all environments.

In the future, I think we should clean up the handling a bit:
 - it seems the grids are explicitly colored in TikZ while the defaults should be used (this is necessary to globally override grid styles in your LaTeX document).

**Comment by miscco** [comment]
The problem with bodeplots if the fucked up way it saves the legends. Therefore we have the problem that all axes get the forget plot option and pgfplots bails. See #855 and #799

**Comment by egeerardyn** [comment]
I knew we had those documented somewhere! Thank you for looking those up.
